Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1031,6 +1054,30 @@ public extension Purchases {
 
 // swiftlint:enable missing_docs
 
+// MARK: - Paywalls
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+public extension Purchases {
+
+    /// Used by `RevenueCatUI` to keep track of ``PaywallEvent``s.
+    func track(paywallEvent: PaywallEvent) async {
+        switch paywallEvent {
+        case let .view(data):
+            self.purchasesOrchestrator.cachePresentedPaywall(data)
+
+        case .close:
+            self.purchasesOrchestrator.clearPresentedPaywall()
```

Comment:
feels like the orchestrator should be the one deciding what to do based on the params

Reply:
Good point, I'll pass the event instead.